lib/sequelize.js

```diff
@@ -911,6 +919,53 @@ class Sequelize {
     return new Utils.Literal(val);
   }
 
+  /**
+   * Creates an object representing sql strings and slots. Slots are values
+   * that will be binded or escaped. Strings and slots will be joined without
+   * spaces.
+   *
+   * @param {...string|Object} items String or slot to be part of the query
+   * @memberof Sequelize
+   *
+   * @returns {Sequelize.composition}
+   */
+  static composition(...items) {
+    return new Utils.Composition(...items.map(item => {
```

Comment:
I've noticed you are performing a lot of array/argument destructuring, how performant is this?
Wouldn't it be better to pass arrays around?

Reply:
Also aren't you directly importing `Composition` already? Might as well use that instead of the `Utils` alias (assuming they are the same).

Reply:
The performance impact only affects `sequelize.query` calls, which behaves similar to `sequelize.literal`. It is not used internally, so any performance penalty is only paid by users that customize queries.

```
// This is the only example that came to my mind
// I believe it can be used for more powerfull things
const field = Model.rawAttributes.field;
Model.findAll({
  where: {
    field: {
      [Op.overlap]: sequelize.composition('int4range(', sequelize.slot(unsafeStart, field), sequelize.slot(unsafeEnd, field), ')')
    }
  },
});
```

Reply:
Take a look at the new review with `composableMixin`, it should be more performant. Now, the classes are `Composition = Object  + composableMixin` and `CompositionMethod = SequelizeMethod + composableMixin`.

I've implemented a little helper `Utils.converComposable` which converts one composable class into another composable class with performance in mind.

Now `sequelize.composition` looks like this:

```
static composition(...items) {
    return Utils.Composition.from(items);
}
```

I could increase performance using `arguments` instead. I'm curious, which is the reason for not using `arguments`?